pkgs/by-name/in/installShellFiles/package.nix

```diff
@@ -0,0 +1,16 @@
+{
+  lib,
+  callPackage,
+  makeSetupHook,
+}:
+
+# See the header comment in ./setup-hook.sh for example usage.
+makeSetupHook {
+  name = "install-shell-files";
+  passthru = {
+    tests = lib.packagesFromDirectoryRecursive {
+      inherit callPackage;
+      directory = ./tests;
+    };
```

Comment:
This is not worth addressing here, but note that this has a minor problem: Due to `packagesFromDirectoryRecursive` recursing for directories, a standard `nix-build -A installShellFiles.tests` wouldn't run any nested ones, because `nix-build` doesn't recurse more than one level. In my experience, recursive attribute sets give rise to various such oddities, which is why I recommend avoiding them. Cc @philiptaron

Reply:
Good to know! There is likely need then for a simple non-recursive variant for these flat use cases.